src/main/java/io/camunda/testing/utils/ProcessInstanceInspections.java

```diff
@@ -0,0 +1,100 @@
+package io.camunda.testing.utils;
+
+import io.camunda.testing.filters.ProcessInstanceRecordStreamFilter;
+import io.camunda.testing.utils.model.InspectedProcessInstance;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class ProcessInstanceInspections {
+
+  private final ProcessInstanceRecordStreamFilter filter;
+
+  public ProcessInstanceInspections(final ProcessInstanceRecordStreamFilter filter) {
+    this.filter = filter;
+  }
+
+  /**
+   * Filters the process instances to only include instances that were started by the given parent
+   * process instance key
+   *
+   * @param key The key of the parent process instance
+   * @return this {@link ProcessInstanceInspections}
+   */
+  public ProcessInstanceInspections startedByProcessInstance(final long key) {
```

Comment:
:x: rename to `withParentProcessInstanceKey` for consistency with assert and also better following of naming conventions here.